src/ProjectService.Business/Commands/Project/GetProjectCommand.cs

```diff
@@ -21,289 +29,340 @@
+
+        if (response.Message.IsSuccess)
+        {
+          return response.Message.Body;
+        }
+        else
+        {
+          _logger.LogWarning("Errors while getting users departments and positions for users {UserIds}. Reason: {Errors}",
+            string.Join(", ", usersIds),
+            string.Join('\n', response.Message.Errors));
         }
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc,
+          "Can not get user's departments and positions for users {UserIds}. Please try again later.",
+          usersIds);
+      }
+
+      errors.Add("Can not get user's departments and positions. Please try again later.");
+
+      return null;
+    }
+
+    public GetProjectCommand(
+      ILogger<GetProjectCommand> logger,
+      IProjectRepository repository,
+      IUserRepository userRepository,
+      IProjectResponseMapper projectResponsMapper,
+      IProjectUserInfoMapper projectUserInfoMapper,
+      IProjectFileInfoMapper projectFileInfoMapper,
+      IDepartmentInfoMapper departmentInfoMapper,
+      IImageInfoMapper imageMapper,
+      IRequestClient<IGetDepartmentsRequest> rcGetDepartments,
+      IRequestClient<IGetUsersDataRequest> usersDataRequestClient,
+      IRequestClient<IGetCompanyEmployeesRequest> rcGetCompanyEmployees,
+      IRequestClient<IGetImagesRequest> rcImages,
+      IConnectionMultiplexer cache)
+    {
+      _logger = logger;
+      _repository = repository;
+      _userRepository = userRepository;
+      _projectResponseMapper = projectResponsMapper;
+      _projectUserInfoMapper = projectUserInfoMapper;
+      _projectFileInfoMapper = projectFileInfoMapper;
+      _departmentInfoMapper = departmentInfoMapper;
+      _imageMapper = imageMapper;
+      _rcGetDepartment = rcGetDepartments;
+      _usersDataRequestClient = usersDataRequestClient;
+      _rcGetCompanyEmployees = rcGetCompanyEmployees;
+      _rcImages = rcImages;
+      _cache = cache;
+    }
+
+    public async Task<OperationResultResponse<ProjectResponse>> Execute(GetProjectFilter filter)
+    {
+      OperationResultResponse<ProjectResponse> response = new();
+      DepartmentData department = null;
+      DbProject dbProject = _repository.Get(filter);
+
+      if (dbProject.DepartmentId.HasValue)
+      {
+        department = await GetDepartment(dbProject.DepartmentId.Value, response.Errors);
+      }
+
+      List<UserData> usersDatas = await GetUsersDatas(dbProject.Users, response.Errors);
+      List<ProjectUserInfo> usersInfo = null;
+      List<Guid> usersIds = dbProject.Users.Select(u => u.UserId).Distinct().ToList();
+
+      if (usersDatas != null && usersDatas.Any())
+      {
+        (List<DepartmentData> departments, List<PositionData> positions) = await GetCompanyEmployee(usersIds, response.Errors);
+        List<ImageInfo> imagesInfos = await GetUserAvatars(usersDatas.Where(u => u.ImageId.HasValue).Select(u => u.ImageId.Value).ToList(), response.Errors);
```

Comment:
maybe you add ?operation:  ....(usersDatas?.Where(u =>....

Reply:
if check that usersData is not null